shared/string.js

```diff
@@ -0,0 +1,27 @@
+String.prototype.isEmpty = function() {
+  return this.trim() === '';
+};
+
+String.prototype.isBlankOrGitOrBit = function() {
+  return /^$|github.com|bitbucket.org/g.test(this);
+};
+
+String.prototype.toLowerHyphenated = function() {
+  return this.toLowerCase().trim().replace(/\s/g, '-');
+};
+
+String.prototype.toCapitalizeUnhyphenated = function() {
+  return this.replace(/-/g, ' ')
+    .replace(/\b\w/g, function(m) {
+      return m.toUpperCase();
+    })
+    .replace(/\s/g, '');
+};
+
+String.prototype.toUpperCaseUnhyphenated = function() {
```

Comment:
But what's the point in creating a function which does `this.replace(/\W+/g, '') ` ? To me it's clear this removes all non-word chars, it's not a complex regex. Should we wrap all regexes in a function?